haystack/nodes/file_converter/pdf.py

```diff
@@ -169,15 +180,27 @@ def _read_pdf(self, file_path: Path, layout: bool, encoding: Optional[str] = Non
                        the content stream order.
         :param encoding: Encoding that overwrites self.encoding and will be passed as `-enc` parameter to `pdftotext`.
                          (See list of available encodings by running `pdftotext -listenc` in the terminal)
+        :param start_page: The page number where to start the conversion
+        :param end_page: The page number where to end the conversion.
         """
         if not encoding:
             encoding = self.encoding
 
-        command = ["pdftotext", "-enc", str(encoding), "-layout" if layout else "-raw", str(file_path), "-"]
+        start_page = start_page or 1
+
+        command = ["pdftotext", "-enc", str(encoding), "-layout" if layout else "-raw", str(start_page)]
```

Comment:
This looks great! One thing is I think you are missing the `"-f"` flag here, but I could be wrong. I'm not too familiar with `pdftotext`

Reply:
This is how a bug get introduced. 🥲